### ISSUE TITLE
[req]: Add Indexer Response Time Statistics

### ISSUE THREAD

**Issue by batuhankrmn** [Jackett/Jackett#15095]
### Is there already a request for your feature?

- [X] I have checked older issues, open and closed

### Is your feature request related to a problem? Please describe.

Yes, when there are a lot of indexers are added to Jackett (like 40+), response time can be long sometimes. In fact, the reason for the long response time is usually not the high number of indexers, but the slow response of a few indexers.

### Describe the solution you'd like

It would be nice to see which indexer took how many seconds to respond to the last manual search query. A "statistics" button can be added at the bottom of the homepage (next to the check for updates button maybe?) When this button is clicked, can we sort the indexer response times for the last query from the longest to the shortest, or as a chart? Prowlarr does this and is very useful for identifying which indexer is causing problems or slowing down queries.

**Comment by garfield69** [comment]
The Jackett Dashboard manual search is a small part of the function of Jackett, whose main purpose is to provide a Torznab service for *arr apps.
while te Dashboard logs already has the response times for each indexer recorded in the log, that data is not cached ATM, so is not at hand for the GUI to provide statistics with.

For bells and whistles we recommend using Prowlarr, its one of that apps primary features. It also has a large cadre of volunteers to support it.
Jackett is simpler. And the active members count is 3 and only one of them part times as a C# dev here.
This is the long way of saying that while I will add this feature to the queue, I don't expect any progress unless some other volunteer decides to contribute.

**Comment by batuhankrmn** [comment]
@garfield69 I understand and appreciate your consideration. I fully agree with Jackett simpler design and I love it actually. So what about only logging indexer response times as single log entry, it will be much easy to read. I guess this would be much more easy to develop compared to develop a button and UI for it. Example log entry:

--Query Completed. Indexers Response Times:
Torrentleech:4.05s,IPTorrents:3.90s,TorrentGalaxy:2.01s....

I'm not a developer and but I assume logging response times in single log entry, and sorting them from longest to shortest would be easy and efficient for Jackett.

**Comment by batuhankrmn** [comment]
> For a search using multiple indexers, the time is given for the search as a whole, rather than per indexer:

Yeah, breaking down to per indexer would be enough IMO. That's what I suggest If it's possible

**Comment by garfield69** [comment]
>but you're saying logs while the user wants it in UI, no?

The OP would be happy with the log version if it is too difficult/complicated to provided it to the GUI

**Comment by batuhankrmn** [comment]
> > but you're saying logs while the user wants it in UI, no?
> 
> The OP would be happy with the log version if it is too difficult/complicated to provided it to the GUI

Exactly, I just look for functionality to check query response times of the indexers and I'm perfectly OK with easier logging way.

**Comment by batuhankrmn** [comment]
> can we sort the indexer response times for the last query from the longest to the shortest, or as a chart?

@mynameisbogdan @garfield69 Thank you for your quick solution and I would like to give feedback on this matter. As I mentioned in my original post, the main reason I wanted this feature was to see the response times of the indexers, sorted from longest to shortest. When I checked the latest version now, I noticed that the response times are displayed in random order. I think it is not very practical to find which indexer among 50 indexers extends the query time. I don't mean to be rude or arrogant, I'm just wondering if it's possible to add this without too much effort.

**Comment by batuhankrmn** [comment]
I agree, it would not be appropriate to design UI based on one person's wishes. For this reason, I suggested sorting from longest response time to shortest in a single log record after the query. Considering that, there will be people who want to not see the response time on the manual search screen at all, maybe they can consider these times as crowded in the UI. If this sorting is done in a single record at the log level, it may be useful for enthusiasts who just want to check. In any case, I appreciate your work, I just wanted to clarify my feature request, it's up to you developers/designers to decide. 🤝🏼

**Comment by ilike2burnthing** [comment]
The vast majority of people who will ever look at the list of indexers for a search results will be looking to find a specific indexer, which is most easily done with an alphabetical list. You'll be one of the few people looking for which took the longest.

As explained in the PR, trying to add it to the logs was spammy.
